### PR TITLE
Add xbee comms

### PR DESCRIPTION
**Pull request by jroelofs**
This adds some XBee synchronization of GameManager states.

### CODE REVIEW

**Comment by jroelofs** [comment]
@garious (if you want to take a look)

File: uwh/xbee_comms.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+from . import messages_pb2
+from .comms import UWHProtoHandler
+
+import time
+
+class XBeeClient(UWHProtoHandler):
+    def __init__(self, mgr):
+        UWHProtoHandler.__init__(self, mgr)
+        self._xbee = XBeeDevice('/dev/tty.usbserial-DN040E8Y', 9600)
```
Comment (jroelofs) [inline uwh/xbee_comms.py]: I plan on exposing the tty and baud rate as parameters soon, and they'll be stored in one of those config files (like the screen size stuff in uwh-refbox).

File: uwh/xbee_comms.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        while xnet.is_discovery_running():
+            time.sleep(0.1)
+
+    def send_raw(self, recipient, data):
+        self._xbee.send_data_async(recipient, data)
+
+    def unicast_send_raw(self, remote, data):
+        self.send_raw(remote, data)
+
+    def multicast_send_raw(self, remotes, data):
```
Comment (jroelofs) [inline uwh/xbee_comms.py]: These are unused, should probably delete them.

File: uwh/xbee_comms.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        self._xbee.send_data_async(recipient, data)
+
+    def listen_loop(self):
+        while True:
+            xbee_msg = self._xbee.read_data()
+            if xbee_msg:
+                self.recv_raw(xbee_msg.remote_device, xbee_msg.data)
+                time.sleep(0.1)
+
+class XBeeServer(UWHProtoHandler):
```
Comment (jroelofs) [inline uwh/xbee_comms.py]: Not sure how/whether to write tests for these two classes. Is there a way to mark tests unsupported if specific hardware isn't available?

Comment (jroelofs) [reply uwh/xbee_comms.py]: (I did try to factor things such that a big portion of it is testable independantly of having actual hardware to test them on).

File: uwh/comms.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    def message_for_msg_kind(self, msg_kind):
+        return { messages_pb2.MessageType_Ping : messages_pb2.Ping,
+                 messages_pb2.MessageType_Pong : messages_pb2.Pong,
+                 messages_pb2.MessageType_GameKeyFrame : messages_pb2.GameKeyFrame
+               }[msg_kind]()
+
+    def pack_message(self, msg_kind, msg):
+        length = msg.ByteSize()
+        assert length < 255 # XBee can't handle huge messages
```
Comment (garious) [inline uwh/comms.py]: You might want to change these asserts to raising exceptions. And then catch them from the top-level loop and log that the unexpected happened.

Comment (jroelofs) [reply uwh/comms.py]: Good idea. That'll make problems non-fatal.

**Pull request merged**